src/services/notehub/NotehubDataProvider.ts

```diff
@@ -106,22 +115,54 @@ export default class NotehubDataProvider implements DataProvider {
 
     const latestSensorEvents = await getAllLatestSensorEvents();
 
-    const simplifiedSensorEvents = uniqBy(
-      flattenDeep(latestSensorEvents)
-        .map((sensorEvent) => ({
-          name: undefined,
-          gatewayUID: sensorEvent.gatewayUID,
-          macAddress: sensorEvent.macAddress.split("#")[0],
-          humidity: sensorEvent.humidity,
-          pressure: sensorEvent.pressure,
-          temperature: sensorEvent.temperature,
-          voltage: sensorEvent.voltage,
-          lastActivity: sensorEvent.lastActivity,
-        }))
-        .filter((addr) => addr !== undefined),
-      "macAddress"
+    const simplifiedSensorEvents = flattenDeep(latestSensorEvents).map(
+      (sensorEvent) => ({
+        name: undefined,
+        gatewayUID: sensorEvent.gatewayUID,
+        macAddress: sensorEvent.macAddress.split("#")[0],
+        humidity: sensorEvent.humidity,
+        pressure: sensorEvent.pressure,
+        temperature: sensorEvent.temperature,
+        voltage: sensorEvent.voltage,
+        count: sensorEvent.count,
+        total: sensorEvent.total,
+        lastActivity: sensorEvent.lastActivity,
+      })
     );
 
+    // merge objects with different defined properties into a single obj
+    const mergeObject = <V>(A: any, B: any): V => {
+      const res: any = {};
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, array-callback-return, @typescript-eslint/no-unsafe-assignment
+      Object.keys({ ...A, ...B }).map((key) => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+        res[key] = B[key] || A[key];
+      });
+      return res as V;
+    };
+
+    // take each event, and make it into a new Map() obj where macAddress is the key
+    // regardless of if there's already a value for that key, run the merge function
+    // set the key and the new merge var as the key, value for the map
```

Comment:
Had to read this a few times to get it. Probably worth a quick grammatical clean up.

Reply:
Cleaned up to be easier to understand